### PR TITLE
Do not raise exception on flusher stop

### LINKED ISSUE #18: SystemExit exception by flusher reported by monitoring tools
I have a FastAPI application with both sentry and logtail. Sentry has a ThreadingIntegration, which reports all unhandled exceptions in threads. And logtail finishes flusher thread with [`sys.exit(0)`](https://github.com/logtail/logtail-python/blob/master/logtail/flusher.py#L71), which is then reported by sentry, which is quite annoying.

If you think it's reasonable to change this logtail behavior to just finish flusher thread silently, I could create a PR for that.

### PR DESCRIPTION
**Pull request by Nnonexistent**
Fixes #18

### CODE REVIEW

File: tests/test_flusher.py [hunk 1]
```diff
@@ -141,4 +140,17 @@ def uploader(frame):
...
+
+    def test_shutdown_dont_raise_exception_in_thread(self):
+        original_excepthook = threading.excepthook
+        threading.excepthook = mock.Mock()
+
+        _, _, fw = self._setup_worker()
+        fw.parent_thread = mock.MagicMock(is_alive=lambda: False)
+        fw.step()
+
+        self.assertFalse(fw.is_alive())
```
Comment (curusarn) [inline tests/test_flusher.py:153]: Same here:
```suggestion
        self.assertFalse(fw.should_run)
```

**Review by curusarn — CHANGES_REQUESTED** [review]
Thanks a lot for the PR! Sorry for the wait! 

Left some minor comments. Otherwise looks good!

**Review by curusarn — APPROVED** [review]

File: tests/test_flusher.py [hunk 2]
```diff
@@ -141,4 +140,17 @@ def uploader(frame):
 
         fw.step()
         self.assertEqual(self.upload_calls, 1)
-        self.assertEqual(mock_exit.call_count, 1)
+        self.assertFalse(fw.should_run)
+
```
Comment (curusarn) [inline tests/test_flusher.py:144]:
```suggestion

# test relies on overriding excepthook which is available from 3.8+
@pytest.mark.skipif(sys.version_info <= (3, 7))
```

File: tests/test_flusher.py [hunk 3]
```diff
@@ -141,4 +140,19 @@ def uploader(frame):
...
+# test relies on overriding excepthook which is available from 3.8+
+@pytest.mark.skipif(sys.version_info <= (3, 7))
```
Comment (curusarn) [inline tests/test_flusher.py:146]:
```suggestion
    # test relies on overriding excepthook which is available from 3.8+
    @pytest.mark.skipif(sys.version_info <= (3, 7))
```

File: logtail/flusher.py [hunk 4]
```diff
@@ -1,8 +1,8 @@
 # coding: utf-8
 from __future__ import print_function, unicode_literals
-import sys
-import time
+
```
Comment (curusarn) [inline logtail/flusher.py:3]:
```suggestion

import sys
```

File: logtail/flusher.py [hunk 5]
```diff
@@ -1,8 +1,9 @@
 # coding: utf-8
 from __future__ import print_function, unicode_literals
+
 import sys
```
Comment (curusarn) [inline logtail/flusher.py:4]:
```suggestion

```

**Comment by Nnonexistent** [comment]
@curusarn I see you've already edited code to your preferences - thanks!

I, hopefully, fixed the skip condition for 3.7 python - pls run the workflow to verify it.

If you need something else done in this PR, please let me know

**Comment by curusarn** [comment]
Thanks! 

I ran the workflow, and it passed. Everything looks good.

I will merge and release a version soon!

**Comment by PetrHeinz** [comment]
@Nnonexistent Thank you for the contribution, I just released it as [v0.2.8](https://github.com/logtail/logtail-python/releases/tag/v0.2.8)

**Pull request merged**